dsp/trackers/langfuse_tracker.py

```diff
@@ -1,18 +1,26 @@
-from typing import Optional, Union, List, Any
+from dataclasses import dataclass
+from typing import Optional, List, Any, NamedTuple
 import httpx
 import logging
 import os
-from langfuse.client import Langfuse, StatefulTraceClient, StatefulSpanClient, StateType
 from dsp.trackers.base import BaseTracker
 
+try:
+    from langfuse.client import Langfuse
+    from langfuse.decorators import observe
+except ImportError:
+    def observe():
```

Comment:
Great idea to switch to the decorator to track any LLM used via dspy instead of relying on the langfuse openai intgeration. I'd suggest to capture token counts and model parameters as these otherwise get lost when making this change: https://langfuse.com/docs/sdk/python/decorators#log-any-llm-call

Reply:
Hit the mark!
If an exception occurs here, it means that the langfuse package is not currently installed. So is it unnecessary for me to obtain the indicator data here? And users can also obtain indicator data through the `history` parameter without installing any tracking tools. 

So I am a little unsure whether to collect indicators here.

Is there anything I haven't considered? Please point it out to me, thank you!

Reply:
Sorry for the confusion, I added the feedback to this line as I recognized that you switched to the `observe` decorator instead of the openai integration when reading this. I was not hinting at that any change at this line is necessary but more in general that you should log out additional llm call information as shown in the docs to maintain these metrics/